### ISSUE TITLE
Compile fails on Linux X86_64 when CURL isn't available

### ISSUE THREAD

**Issue by volumetricsteve** [yquake2/yquake2#381]
In the Makefile, I see 

`WITH_CURL:=yes`

my system/os (MX Linux) doesn't make it clear which library I need for development with curl.  For simplicity's sake, I changed that yes to a no and it gets all the way down to the linking phase on the client and I get:

```
===> LD release/quake2
build/client/src/client/cl_download.o: In function `CL_RequestNextDownload':
/test/quake2-7.40/src/client/cl_download.c:508: undefined reference to `CL_HTTP_EnableGenericFilelist'
collect2: error: ld returned 1 exit status
Makefile:948: recipe for target 'release/quake2' failed
make[1]: *** [release/quake2] Error 1
make[1]: Leaving directory '/test/quake2-7.40'
Makefile:366: recipe for target 'client' failed
make: *** [client] Error 2
```
 

I can probably find a way to work around it but I felt like this was worth reporting.

Edit:
I've already semi-worked around this by just downloading the curl source (presented in linux from scratch) and compiling it && putting it all into an extra library path on my system.  I just updated all the CFLAGS, LD_LIBRARY_PATH etc etc and set "WITH_CURL:=yes" again, now it builds fine.

I suspect there's some curl code that's not switching off when that's set to 'no'

What is CURL used for in YQ2 anyway?

**Comment by Yamagi** [comment]
> What is CURL used for in YQ2 anyway?
For HTTP downloads, compatible with r1q2 and q2pro.

The bug was already fixed in 32d9119cffd4d66b3c0962be21c04c6c8fc52911

**Comment by volumetricsteve** [comment]
Oh ok cool, thank you
